test/Spec/MockContract.hs

```diff
@@ -146,15 +160,32 @@ addr1 = unsafeSerialiseAddress Mainnet (Ledger.pubKeyHashAddress paymentPkh1 Not
 addr2 = unsafeSerialiseAddress Mainnet (Ledger.pubKeyHashAddress paymentPkh2 Nothing)
 addr3 = unsafeSerialiseAddress Mainnet (Ledger.pubKeyHashAddress paymentPkh3 Nothing)
 
-toPubKey :: SigningKey PaymentKey -> PubKey
-toPubKey = Ledger.toPublicKey . fromRight (error "Impossible happened") . Files.fromCardanoPaymentKey
+skeyToPubKey :: SigningKey PaymentKey -> PubKey
+skeyToPubKey =
+  Ledger.toPublicKey
+    . Files.unDummyPrivateKey
+    . fromRight (error "Impossible happened")
```

Comment:
How impossible are we talking, maybe give something a little more meaningful, since getting "impossible happened" in an error would not be super helpful

Reply:
Yeah, sorry about that. Fixed it.